proxy-robots-module/src/main/java/orwell/proxy/ProxyRobots.java

```diff
@@ -103,12 +104,15 @@ protected void sendRegister() {
      */
     protected void sendServerRobotStates() {
         for (final IRobot robot : robotsMap.getRegisteredRobots()) {
-            final byte[] zmqServerRobotState = robot.getServerRobotStateBytes_And_ClearDelta();
-            if (null != zmqServerRobotState) {
+            final RobotElementStateVisitor stateVisitor = new RobotElementStateVisitor();
+            robot.accept(stateVisitor);
+
+            final byte[] serverRobotStateBytes = stateVisitor.getServerRobotStateBytes();
```

Comment:
I am not really happy with extracting the array from the object and computing the ZMQ message here.